cupy/math/sumprod.py

```diff
@@ -147,8 +152,15 @@ def _proc_as_batch(proc, x, axis):
     return r.reshape(s).transpose(revert)
 
 
-def _cum_core(a, axis, dtype, out, kern, batch_kern):
+def _cum_core(a, axis, dtype, out, kern, batch_kern, *, op=None):
     a = cupy.asarray(a)
+
+    if cupy.cuda.cub_enabled:
+        # result will be None if the scan is not compatible with CUB
+        result = cub.cub_scan(a, op, axis, dtype, out)
+        if result is not None:
+            return result
+
```

Comment:
Does this work for inplace?

Reply:
It might, let me look into this.

Reply:
See https://github.com/cupy/cupy/pull/2919#issuecomment-575968811